### PR TITLE
feat(common): sentry release control

### PR DESCRIPTION
**Pull request by mcdurdin**
This centralises linking the current commit to the tagged version. It will be called automatically when we increment the version, just before a build is done. This allows us to tag the built projects with the version in question in Sentry.

A future PR will provide debug information file uploads that link into this process. Thus the stubs in sentry-control.sh.

See https://docs.sentry.io/workflow/releases/?platform=native#configure-sdk for how to configure our usage to reference the release. The release ID we use will be `release-$VERSION_WITH_TAG` which matches the tagging we do in the repository, for consistency.

Note that local dev builds will always have `-local` appended and this reflects the potential mismatch between our local repo and the origin repo. Currently, only builds on master, beta and stable branches will receive a release tag. Given our current development patterns, I think this is sufficient -- we don't do enough testing on PR or feature branches at present to warrant the additional meta-releases being sent through. This should be easy enough to add if we want to in the future.

### CODE REVIEW

File: resources/build/sentry-control.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# This script centralises sentry symbol uploads and releases
+# Note: We must install sentry-cli on the build agents
+#
+
+
+## TODO: centralise debug information file (dif) uploads
+
+
+function isSentryConfigured() {
+  if [ -z ${SENTRY_AUTH_TOKEN-} ] || [ -z ${SENTRY_ORG} ] || [ -z ${SENTRY_URL} ]; then
```
Comment (darcywong00) [inline resources/build/sentry-control.sh]: Is it 
`${SENTRY_AUTH_TOKEN-}` or 
`${SENTRY_AUTH_TOKEN}`?

File: resources/build/sentry-control.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+function makeSentryRelease() {
+  if isSentryConfigured; then
+    if isSentryCliAvailable; then
+      local SENTRY_RELEASE_VERSION="release-$VERSION_WITH_TAG"
+      echo "Making a Sentry release for tag $SENTRY_RELEASE_VERSION"
+      sentry-cli releases new -p keyman-android -p keyman-developer -p keyman-ios -p keyman-linux -p keyman-mac -p keyman-web -p keyman-windows $SENTRY_RELEASE_VERSION
+      # Testing: sentry-cli releases new -p keyman-windows $SENTRY_RELEASE_VERSION
+
+      echo "Setting commits for release tag $SENTRY_RELEASE_VERSION"
+      sentry-cli releases set-commits --auto $SENTRY_RELEASE_VERSION
```
Comment (darcywong00) [inline resources/build/sentry-control.sh]: Would you [finalize](https://docs.sentry.io/cli/releases/#finalizing-releases) the release?

File: resources/build/sentry-control.sh (hunk 1 above)
Comment (mcdurdin) [reply resources/build/sentry-control.sh]: The `-` is right. The other two tests need it too! 😊

Comment (mcdurdin) [reply resources/build/sentry-control.sh]: https://stackoverflow.com/a/40630951/1836776

File: resources/build/sentry-control.sh (hunk 2 above)
Comment (mcdurdin) [reply resources/build/sentry-control.sh]: It's on my agenda. Will consider how I want to do this over the next week or so

Comment (mcdurdin) [reply resources/build/sentry-control.sh]: It's on my agenda. Will consider how I want to do this over the next week or so. In a new PR though.

**Pull request merged**